src/test/socket/getsockname/test_getsockname.rs

```diff
@@ -461,6 +466,91 @@ fn test_bound_socket(domain: libc::c_int, sock_type: libc::c_int) -> Result<(),
     Ok(())
 }
 
+/// Test getsockname using an autobound socket.
+fn test_autobound_socket(domain: libc::c_int, sock_type: libc::c_int) -> Result<(), String> {
+    let fd = unsafe { libc::socket(domain, sock_type, 0) };
+    assert!(fd >= 0);
+
+    // the sockaddr that we will bind to
+    let (bind_addr, bind_addr_len) = match domain {
+        libc::AF_INET => (
+            SockAddr::Inet(libc::sockaddr_in {
+                sin_family: libc::AF_INET as u16,
+                sin_port: 0u16.to_be(),
+                sin_addr: libc::in_addr {
+                    s_addr: libc::INADDR_LOOPBACK.to_be(),
+                },
+                sin_zero: [0; 8],
+            }),
+            std::mem::size_of::<libc::sockaddr_in>() as u32,
+        ),
+        libc::AF_UNIX => (
+            SockAddr::Unix(libc::sockaddr_un {
+                sun_family: libc::AF_UNIX as u16,
+                sun_path: [0i8; 108],
+            }),
+            2,
+        ),
+        _ => unimplemented!(),
+    };
+
+    // bind to the sockaddr
+    let rv = unsafe { libc::bind(fd, bind_addr.as_ptr(), bind_addr_len) };
+    assert_eq!(rv, 0);
+
+    // fill the sockaddr with dummy data
+    let addr = match domain {
+        libc::AF_INET => SockAddr::dummy_init_inet(),
+        libc::AF_UNIX => SockAddr::dummy_init_unix(),
+        _ => unimplemented!(),
+    };
+
+    // getsockname() may mutate addr and addr_len
+    let mut args = GetsocknameArguments {
+        fd: fd,
+        addr: Some(addr),
+        addr_len: Some(addr.ptr_size()),
+    };
+
+    test_utils::run_and_close_fds(&[fd], || check_getsockname_call(&mut args, None))?;
+
+    let expected_addr_len = match domain {
+        libc::AF_INET => std::mem::size_of::<libc::sockaddr_in>() as u32,
+        // 2 + 1 + 5
```

Comment:
Not clear to me where these come from. Possibly overkill, but one way to spell it out is something like: 

```
libc:: AF_UNIX => {
  let foo_len = 2;
  let bar_len = 1;
  let baz_len = 5;
  foo_len + bar_len + baz_len
}
```

Reply:
I added a comment instead explaining where they came from.